### PR TITLE
Add cypress for e2e cli testing (meta)

### PR DESCRIPTION
**Pull request by wafuwafu13**
Related: https://github.com/blitz-js/blitz/issues/1767

### What are the changes and their implications?

>We currently have a few blitz new tests, but not a complete e2e test for the new app.
>First person needs to add cypress and very basic initial test.

```
~/desktop/blitz/packages/cli
$ npm run cypress:open
```
<details>
<summary>Result</summary>

スクリーンショット 2021-02-02 16 57 13

click `basic.ts`

スクリーンショット 2021-02-02 16 59 15

</details>

```
~/desktop/blitz/packages/cli
$ npm run cypress:run
```

<details>
<summary>Result</summary>

```
> e2e@0.0.0 cypress:run /Users/tagawahirotaka/Desktop/blitz/packages/cli/test/e2e
> cypress run

Missing baseUrl in compilerOptions. tsconfig-paths will be skipped

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    6.4.0                                                                              │
  │ Browser:    Electron 87 (headless)                                                             │
  │ Specs:      1 found (basic.ts)                                                                 │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  basic.ts                                                                        (1 of 1)


  Google Test
    ✓ Shows Google (1195ms)


  1 passing (3s)


  (Results)

┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     2 seconds                                                                        │
  │ Spec Ran:     basic.ts                                                                         │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Video)

  -  Started processing:  Compressing to 32 CRF
  -  Finished processing: /Users/tagawahirotaka/Desktop/blitz/packages/cli/test/e2e/c     (1 second)
                          ypress/videos/basic.ts.mp4


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  basic.ts                                 00:02        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:02        1        1        -        -        -
```

</details>

### Checklist

- [x] Changes covered by tests (tests added if needed)
- [x] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes

### CODE REVIEW

**Comment by flybayer** [comment]
Nice, this is a great start!!

Would you be able to add some actual CLI tests as described in https://github.com/blitz-js/blitz/issues/1767?

**Comment by wafuwafu13** [comment]
Thank you!
I'll do it :)
So could you merge this PR ?

File: packages/cli/test/e2e/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "scripts": {
+    "cypress:open": "cypress open",
+    "cypress:run": "cypress run"
+  },
+  "private": true,
+  "dependencies": {},
+  "devDependencies": {
+    "cypress": "^6.4.0"
+  }
+}
```
Comment (flybayer) [inline packages/cli/test/e2e/package.json:14]: Ok, yes, one thing before merging:

- Move `cypress` dependency to the very root package.json of the repo. 
- Move these e2e test scripts to `packages/cli/package.json`
- Remove this package.json

Comment (wafuwafu13) [reply packages/cli/test/e2e/package.json]: https://github.com/blitz-js/blitz/pull/1846/commits/de466050fa31458c6e4835cd5f4a71cd00b09158

If I move these e2e test scripts to packages/cli/package.json, I have to move `cypress` folder and `cypress.json`. (`packages/cli/test/e2e/cypress/`  => `packages/cli/cypress`).
Is it Ok?

File: packages/cli/cypress/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (wafuwafu13) [inline packages/cli/cypress/tsconfig.json:1]: ref: https://docs.cypress.io/guides/tooling/typescript-support.html#Configure-tsconfig-json

File: packages/cli/test/e2e/package.json (hunk 1 above)
Comment (wafuwafu13) [reply packages/cli/test/e2e/package.json]: https://github.com/blitz-js/blitz/pull/1846/commits/8dda5cc7f3bbd1af0ab15f5563076f8b83f719a4
I resolved this problem by setting `cypress.json`

File: packages/cli/test/e2e/cypress/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "lib": ["dom", "es5"],
+    "types": ["cypress"]
+  }
+}
```
Comment (wafuwafu13) [inline packages/cli/test/e2e/cypress/tsconfig.json:7]: Ref: https://docs.cypress.io/guides/tooling/typescript-support.html#Configure-tsconfig-json

**Comment by flybayer** [comment]
Sweet, thanks!

@all-contributors add @wafuwafu13 for code

**Comment by flybayer** [comment]
@all-contributors add @wafuwafu13 for test

**Review by flybayer — APPROVED** [review]

**Review by flybayer — APPROVED** [review]

**Pull request merged**